test/src/test/java/org/corfudb/runtime/collections/SMRMultiLogunitTest.java

```diff
@@ -187,15 +187,14 @@ public void transactionalManyWritesThenRead() throws TransactionAbortedException
         for (int i=0; i < numKeys; i++) {
             String key = "key" + String.valueOf(i);
             String val = testMap.get(key);
-            if (val != null) {
-                if (i == 0) {
-                    int underscore = val.indexOf("_");
-                    assertNotEquals(-1, underscore);
-                    base = val.substring(0, underscore);
-                    System.out.println("base is " + base);
-                }
+            if (val != null && i == 0) {
+                int underscore = val.indexOf("_");
+                assertNotEquals(-1, underscore);
+                base = val.substring(0, underscore);
+                System.out.println("base is " + base);
             }
-            assertEquals(true, val.contains(base));
+            assert val != null;
```

Comment:
I think if this assertion is moved before the if condition, then it doesn't need to be checked a second time and the condition can be simplified. You can also use `assertNotNull`.

Reply:
Done, thanks!